timely_beliefs/beliefs/classes.py

```diff
@@ -506,38 +544,15 @@ def __init__(  # noqa: C901
 
             super().__init__(*args, **kwargs)
 
-            # Set up empty frame with columns, indices and attributes
-            if len(args) == 0:
-                self.reset_index(inplace=True)
-                if "belief_horizon" in self and "belief_time" not in self:
-                    indices = [
-                        "belief_horizon" if index == "belief_time" else index
-                        for index in indices
-                    ]
-                if "event_end" in self and "event_start" not in self:
-                    indices = [
-                        "event_end" if index == "event_start" else index
-                        for index in indices
-                    ]
-                for col in columns + indices:
-                    self[col] = None
-                self.set_index(indices, inplace=True)
-                self.sensor = sensor
-                self.event_resolution = sensor.event_resolution if sensor else None
-                return
-            if isinstance(args[0], (pd.DataFrame, pd.Series)) and not isinstance(
+            if len(args) == 0 or (
+                self.empty
+                and isinstance(args[0], (pd.DataFrame, pd.Series))
+                and not isinstance(args[0], BeliefsDataFrame)
+            ):
+                set_columns_and_indices_for_empty_frame(self, columns, indices)
+            elif isinstance(args[0], (pd.DataFrame, pd.Series)) and not isinstance(
```

Comment:
This condition (repeated twice) could be placed nicer in a small helper function, e.g `is_pandas_datastructure(detect_tb_structure=False)`